### PR TITLE
Improvements for Variant_Classification and Amino_Acid_Change columns validation

### PR DESCRIPTION
**Pull request by pieterlukasse**
# What? Why?
Improvements for Variant_Classification and Amino_Acid_Change columns validation. These new validations should help prevent problems that lead to missing COSMIC mappings/annotations in the DB and respective UI.

Changes proposed in this pull request:
- Validation script will now give a WARNING when the `Variant_Classification` is not one of the values listed in https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification. The value `Unknown` is also allowed as a special case.
- The script will also give an ERROR when a white space is found in the `Amino_Acid_Change` or `HGVSp_Short` columns. The loader (java) layer does not strip this white space, so the validation script now also does the same.

### CODE REVIEW

**Comment by pieterlukasse** [comment]
Notify: @xdidiz

**Comment by pieterlukasse** [comment]
TODO: add warnings to documentation warning users that it is case sensitive.

File: core/src/main/scripts/importer/validateData.py [hunk 1]
```diff
@@ -981,6 +981,22 @@ class MutationsExtendedValidator(Validator):
...
+    # MAF values for Variant_Classification column
+    # from https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification + Unknown:
+    VARIANT_CLASSIFICATION_VALUES = [
+       'Frame_Shift_Del',
+       'Frame_Shift_Ins',
+       'In_Frame_Del',
+       'In_Frame_Ins',
+       'Missense_Mutation',
+       'Nonsense_Mutation',
+       'Splice_Site',
```
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: While testing with SKCM TCGA, `Splice_Region` also occurred quite often:
`Splice_Region`: 2721
`Splice_Region`: 2895

Apparently it's unofficial, see https://github.com/mskcc/vcf2maf/issues/88. cBioPortal has no support for Splice_Region, so it's okay to not include it here.

Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: While testing with SKCM TCGA, `Splice_Region` also occurred quite often:
`Splice_Region`: 2721
`Splice_Region`: 2895

Apparently it's unofficial, see https://github.com/mskcc/vcf2maf/issues/88. cBioPortal has no support for Splice_Region, so it's okay to not include it here.

Comment (pieterlukasse) [inline core/src/main/scripts/importer/validateData.py]: @sandertan thanks for the extra test on SKCM. If I understand @ckandoth's comments correctly in the ticket you mentioned, `Splice_Region` and `Splice_Site` are useful information and will remain in the files. We might as well just add them to the list and the documentation. I will add them, unless @ckandoth has any concerns against this.

**Comment by pieterlukasse** [comment]
@sandertan can you test again for the newly added `Splice_Region` and `Splice_Site` (also how these values are displayed in the portal)?

File: core/src/main/scripts/importer/validateData.py [hunk 2]
```diff
@@ -981,8 +981,10 @@ class MutationsExtendedValidator(Validator):
 
     NULL_AA_CHANGE_VALUES = ('', 'NULL', 'NA')
 
+    # extra unofficial Variant classification values from https://github.com/mskcc/vcf2maf/issues/88:
+    EXTRA_VARIANT_CLASSIFICATION_VALUES = ['Splice_Region', 'Splice_Site']
```
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: @pieterlukasse Sorry I was not clear; `Splice_Site` is an official classification, but `Splice_Region` is not. Can you remove `Splice_Site` from `EXTRA_VARIANT_CLASSIFICATION_VALUES`? It is already is still (correctly) in `VARIANT_CLASSIFICATION_VALUES`.

File: docs/File-Formats.md [hunk 3]
```diff
@@ -519,7 +519,7 @@ A minimal mutation annotations file can contain just three of the MAF columns pl
 
 * **Hugo_Symbol**: (MAF column) A [HUGO](http://www.genenames.org/) gene symbol.
 * **Tumor_Sample_Barcode**: (MAF column) This is the sample ID as listed in the clinical data file.
-* **Variant_Classification**: (MAF column) Translational effect of variant allele, e.g. Missense_Mutation, Silent, etc. cBioPortal skips the following types during the import: _Silent_, _Intron_, _3'UTR_, _3'Flank_, _5'UTR_, _5'Flank_, _IGR_ and _RNA_
+* **Variant_Classification**: (MAF column) Translational effect of variant allele. Allowed values (from [Mutation Annotation Format page](https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification)): `Frame_Shift_Del, Frame_Shift_Ins, In_Frame_Del, In_Frame_Ins, Missense_Mutation, Nonsense_Mutation, Silent, Splice_Site, Translation_Start_Site, Nonstop_Mutation, 3'UTR, 3'Flank, 5'UTR, 5'Flank, IGR1 , Intron, RNA, Targeted_Region, De_novo_Start_InFrame, De_novo_Start_OutOfFrame`. cBioPortal skips the following types during the import: `Silent, Intron, 3'UTR, 3'Flank, 5'UTR, 5'Flank, IGR and RNA`. :warning: the values should be in the correct case. E.g. `missense_mutation` is not allowed, while `Missense_Mutation` is.
```
Comment (sandertan) [inline docs/File-Formats.md]: Can we add Splice_Region here as unofficial classification?

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (sandertan) [inline core/src/main/scripts/importer/validateData.py]: We might want to add some styling or functionality for `Splice_Region`. Maybe label it as **Splice** ?

screen shot 2017-05-26 at 11 19 23

`Splice_Site` looks find, and like this:

screen shot 2017-05-26 at 11 21 06

File: core/src/main/scripts/importer/validateData.py (hunk 2 above)
Comment (pieterlukasse) [inline core/src/main/scripts/importer/validateData.py]: ok, done.

File: docs/File-Formats.md (hunk 3 above)
Comment (pieterlukasse) [inline docs/File-Formats.md]: Regarding the styling: could be a good idea, but out of the scope of this PR. Can you log a ticket? 
Regarding documentation: good point. I updated the documentation now.

**Comment by sandertan** [comment]
Nice changes.

**Comment by ckandoth** [comment]
Glad you're handling `Splice_Region`. No concerns. Let me know if you have any other qns.

**Comment by sandertan** [comment]
I created an issue regarding what to do with Splice_Region in the mutation view: https://github.com/cBioPortal/cbioportal/issues/2497

**Comment by pieterlukasse** [comment]
rebase to `hotfix` and squashed second commit into first one (now only 2 commits in total).

**Pull request merged**